mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -489,6 +540,14 @@ def _compare_metrics(metric_a, metric_b):
             return (metric_a.step, metric_a.timestamp, metric_a.value) > \
                    (metric_b.step, metric_b.timestamp, metric_b.value)
 
+        if logged_metrics is not None:
+            for m in logged_metrics:
+                if logged_metric is None or _compare_metrics(m, logged_metric):
+                    logged_metric = m
+
+        if logged_metric is None:
+            return
```

Comment:
`None` metrics should be an error case and should have been handled in the caller function

Reply:
The method will now check if an error should be raised